src/screens/dialogs/upload-file-dialog.tsx

```diff
@@ -0,0 +1,82 @@
+import { useContext } from "react";
+import toast from "react-hot-toast";
+import { uploadfileService } from "../../services/files/upload-file.service";
+import { Dialog } from "../../components/Dialog";
+import {
+  FilesDialogsContext,
+  AuthContext,
+  AVAILABLE_DIALOGS,
+  FilesContext
+} from "../../context/index";
+
+interface INewFile {
+  isFile: boolean;
+  isReady: boolean;
+  name: string;
+  size: number;
+  uuid: string;
+}
+
+export const UploadFileDialog = () => {
+  // Dialog state
+  const { closeDialog, dialogsVisibilityState } =
+    useContext(FilesDialogsContext);
+  const isOpen = dialogsVisibilityState[AVAILABLE_DIALOGS.UPLOAD_FILE];
+  const { addFile, currentDirectory } = useContext(FilesContext);
+  const { session } = useContext(AuthContext);
+
+  const uploadFile = async () => {
+    const fileInput = document.querySelector<HTMLInputElement>("#myfile");
+    if (fileInput?.files?.length) {
+      const fileList = fileInput.files;
+      for (let i = 0; i < fileList.length; i++) {
+        const file = fileList[i];
+
+        const token = session!.token;
+        const uploadRequest = {
+          fileContent: file,
+          fileName: file.name,
+          location: currentDirectory || "",
+          token
+        };
+        const response = await uploadfileService(uploadRequest);
+
+        if (response.success) {
+          const newFile: INewFile = {
+            isFile: true,
+            isReady: true,
+            name: file.name,
+            size: 0,
+            uuid: response.fileUUID!
+          };
+          toast.success(`The file ${file.name} has been uploaded successfully`);
+          addFile(newFile);
+          closeDialog(AVAILABLE_DIALOGS.UPLOAD_FILE);
+        } else {
+          toast.error(response.msg);
+        }
+      }
+    }
+  };
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={() => closeDialog(AVAILABLE_DIALOGS.UPLOAD_FILE)}
+      title="Select the files to upload"
+    >
+      <input
+        type="file"
+        id="myfile"
+        aria-label="Select the files"
+        className="w-full rounded-lg border p-2"
+      />
+      <button
```

Comment:
Please, show a loader or something else to indicate that the files are being uploaded. Right now the modal wouldn't close until the files are completely uploaded and the button will be active while the files are being uploaded, so the users can send **multiple request with heavy files** which can affect our bandwidth. 

I think one option could be create a `loading` state and shown an spinner among the text in the button and add a conditional to the `uploadFile` function: 

```typescript
if(loading) return; 
```

![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/7337d365-143e-412c-befb-faed280f42a6)